src/plat/spike/machine/hardware.c

```diff
@@ -140,24 +140,35 @@ ackInterrupt(irq_t irq)
 }
 
 static inline uint64_t get_cycles(void)
-#if __riscv_xlen == 32
 {
+    UNUSED unsigned long mtvec;
+
+    if (config_set(CONFIG_SEL4_RV_MACHINE)) {
+        /* Trap to riscv-pk to get time */
+        mtvec = read_csr_env(tvec);
+        write_csr(mtvec, 0x80000004ul);
```

Comment:
I find this a little confusing. On the one hand you `read_csr_env`, but then you explicitly write to mtvec, and then later on you put this value that you read explicitly into mtvec. I realize that all this code is guarded by `CONFIG_SEL4_RV_MACHINE`, but you should still be consistent if you use register abstractions or not